tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/SyncMethodInvokerIntegration.cs

```diff
@@ -0,0 +1,78 @@
+// <copyright file="SyncMethodInvokerIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Wcf
+{
+    /// <summary>
+    /// System.ServiceModel.Dispatcher.SyncMethodInvoker calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "System.ServiceModel",
+        TypeName = "System.ServiceModel.Dispatcher.SyncMethodInvoker",
+        MethodName = "Invoke",
+        ReturnTypeName = ClrNames.Object,
+        ParameterTypeNames = new[] { ClrNames.Object, "System.Object[]", "System.Object[]&" },
+        MinimumVersion = "4.0.0",
+        MaximumVersion = "4.*.*",
+        IntegrationName = WcfCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class SyncMethodInvokerIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="instanceArg">RequestContext instance</param>
+        /// <param name="inputs">Input arguments</param>
+        /// <param name="outputs">Output arguments</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object instanceArg, object[] inputs, out object[] outputs)
+        {
+            outputs = default;
```

Comment:
This is ok, just as a comment, you can also use `ref` (`out` is just an special `ref`) that way, you can avoid setting the `outputs` variable.

Reply:
Followed your suggestion and made this a `ref`. Thanks for the suggestion!